diff -urN old/src/metadata/libmp4v2_handler.cc new/src/metadata/libmp4v2_handler.cc
--- old/src/metadata/libmp4v2_handler.cc	2012-04-04 23:17:46.000000000 +0200
+++ new/src/metadata/libmp4v2_handler.cc	2012-04-05 00:22:31.000000000 +0200
@@ -65,29 +65,25 @@
 static void addMetaField(metadata_fields_t field, MP4FileHandle mp4, Ref<CdsItem> item)
 {
     String value;
-    char*  mp4_retval = NULL;
-    u_int16_t track;
-    u_int16_t total_tracks;
- 
     Ref<StringConverter> sc = StringConverter::i2i();
     
+    const MP4Tags* new_tags = MP4TagsAlloc();
+    MP4TagsFetch(new_tags, mp4);
     switch (field)
     {
         case M_TITLE:
-            MP4GetMetadataName(mp4, &mp4_retval);
+            value = new_tags->name;
             break;
         case M_ARTIST:
-            MP4GetMetadataArtist(mp4, &mp4_retval);
+            value = new_tags->artist;
             break;
         case M_ALBUM:
-            MP4GetMetadataAlbum(mp4, &mp4_retval);
+            value = new_tags->album;
             break;
         case M_DATE:
-            MP4GetMetadataYear(mp4, &mp4_retval);
-            if (mp4_retval)
+            value = new_tags->releaseDate;
+            if (value.length() > 0)
             {
-                value = mp4_retval;
-                free(mp4_retval);
                 if (string_ok(value))
                     value = value + "-01-01";
                 else
@@ -95,34 +91,31 @@
             }
             break;
         case M_GENRE:
-            MP4GetMetadataGenre(mp4, &mp4_retval);
+            value = new_tags->genre;
             break;
         case M_DESCRIPTION:
-            MP4GetMetadataComment(mp4, &mp4_retval);
+            value = new_tags->comments;
             break;
         case M_TRACKNUMBER:
-            MP4GetMetadataTrack(mp4, &track, &total_tracks);
-            if (track > 0)
+            if (new_tags->track)
             {
-                value = String::from(track);
-                item->setTrackNumber((int)track);
+                value = String::from(new_tags->track->index);
+                item->setTrackNumber((int)new_tags->track->index);
             }
             else
+			{
+			    MP4TagsFree( new_tags );
                 return;
+            }
             break;
         default:
+			MP4TagsFree( new_tags );
             return;
     }
 
-    if ((field != M_DATE) && (field != M_TRACKNUMBER) && 
-        (mp4_retval))
-    {
-        value = mp4_retval;
-        free(mp4_retval);
-    }
-    
+	MP4TagsFree( new_tags );
     value = trim_string(value);
-    
+
     if (string_ok(value))
     {
         item->setMetadata(MT_KEYS[field].upnp, sc->convert(value));
@@ -190,21 +183,24 @@
         }
 
 #if defined(HAVE_MAGIC)
-        u_int8_t *art_data;
+        void *art_data;
         u_int32_t art_data_len;
         String art_mimetype;
+
+        const MP4Tags* new_tags = MP4TagsAlloc();
+        MP4TagsFetch(new_tags, mp4);
+        const MP4TagArtwork* art = new_tags->artwork;
+        art_data = art->data;
+        art_data_len = art->size;
 #ifdef HAVE_MP4_GET_METADATA_COVER_ART_COUNT
-        if (MP4GetMetadataCoverArtCount(mp4) && 
-            MP4GetMetadataCoverArt(mp4, &art_data, &art_data_len))
-#else
-            MP4GetMetadataCoverArt(mp4, &art_data, &art_data_len);
+        if (new_tags->artworkCount && art_data_len > 0) 
 #endif
         {
             if (art_data)
             {
                 try
                 {
-                    art_mimetype = ContentManager::getInstance()->getMimeTypeFromBuffer((void *)art_data, art_data_len);
+                    art_mimetype = ContentManager::getInstance()->getMimeTypeFromBuffer(art_data, art_data_len);
                     if (!string_ok(art_mimetype))
                         art_mimetype = _(MIMETYPE_DEFAULT);
 
@@ -226,6 +222,7 @@
             }
         }
 #endif
+        MP4TagsFree(new_tags);
         MP4Close(mp4);
     }
     catch (Exception ex)
@@ -253,9 +250,15 @@
     if (!MP4GetMetadataCoverArtCount(mp4))
         throw _Exception(_("LibMP4V2Handler: resource has no album art information"));
 #endif
-    u_int8_t *art_data;
+    void *art_data;
     u_int32_t art_data_len;
-    if (MP4GetMetadataCoverArt(mp4, &art_data, &art_data_len))
+
+    const MP4Tags* new_tags = MP4TagsAlloc();
+    MP4TagsFetch(new_tags, mp4);
+    const MP4TagArtwork* art = new_tags->artwork;
+    art_data = art->data;
+    art_data_len = art->size;
+    if (art)
     {
         if (art_data)
         {
@@ -265,10 +268,10 @@
             return h;
         }
     }
-        
+
     throw _Exception(_("LibMP4V2Handler: could not serve album art "
-                           "for file") + item->getLocation() + 
-                           " - embedded image not found");
+            "for file") + item->getLocation() + 
+        " - embedded image not found");
 }
 
 #endif // HAVE_LIBMP4V2
De binära filerna old/src/metadata/.libmp4v2_handler.cc.swp och new/src/metadata/.libmp4v2_handler.cc.swp skiljer
